crates/sui-graphql-rpc/schema.graphql

```diff
@@ -0,0 +1,823 @@
+# Copyright (c) Mysten Labs, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+# Open Questions
+# --------------
+# - Rules for nullabilit/non-nullability
+# - Pattern for multi-gets
+# - Encoding RPC errors
+# - `type` (vs `scalar`) for Move Types
+# - Pattern for upcasting and downcasting
+
+# Check Completeness Against Existing APIs
+# -------------
+# DONE Read
+# DONE - Objects
+# DONE - Transactions
+# DONE - Checkpoints
+# DONE - Events
+# DONE - ProtocolConfig
+# DONE - Chain Identifier
+# DONE Coin
+# DONE Extended
+# DONE Governance
+# DONE Indexer
+# DONE Move Utils
+# TODO Write
+
+# DONE Port designs from Notion
+# DONE - Display
+# DONE - SuiNS
+# DONE Data Consistency API
+# TODO Flesh out Connections
+# TODO - TotalTransactionBlocks for the PTB connection
+
+schema {
+  query: Query
+  subscription: Subscription
+}
+
+type Query {
+  # First four bytes of the network's genesis checkpoint digest
+  # (uniquely defines the network)
+  chainIdentifier: String!
+
+  # First and last checkpoint that the RPC can answer consistent
+  # queries for.
+  consistentRange: ConsistentRange
+
+  owner(address: SuiAddress!): Owner
+  object(address: SuiAddress!, version: BigInt): Object
+  address(address: SuiAddress!): Address
+
+  epoch(epochId: Int): Epoch
+  protocolConfig(protocolVersion: Int): ProtocolConfig
+
+  checkpoint(filter: CheckpointFilter): Checkpoint
+  checkpoints(order: Order!): CheckpointConnection
+
+  transactionBlock(digest: Bytes!): TransactionBlock
+  transactionBlocks: TransactionBlockConnection
+
+  coinMetadata(coinType: String!): CoinMetadata
+
+  queryObjects(query: ObjectQuery): ObjectQueryConnection
+  queryEvents(filter: EventFilter, order: Order!): EventQueryConnection
+  queryTransactionBlocks(
+    filter: TransactionBlockFilter,
+    order: Order!,
+  ): TransactionBlockQueryConnection
+
+  resolveNameServiceAddress(name: String!): Address
+
+  networkMetrics: NetworkMetrics
+  moveCallMetrics: MoveCallMetrics
+  checkpointAddressMetrics(checkpoint: BigInt): AddressMetric
+  allEpochAddressMetrics(order: Order!): AddressMetricConnection
+}
+
+type Subscription {
+  events(filter: EventFilter): Event
+  transactions(filter: TransactionBlockFilter): TransactionBlock
+}
+
+# String containing 32B hex-encoded address
+scalar SuiAddress
+
+# String representation of an arbitrary width, possibly signed integer
+scalar BigInt
+
+# String containing Base58-encoded binary data.
+scalar Bytes
```

Comment:
I think you should name this type better, stating that its Base58 encoded in the name potentially. base58 is not a good format to transmit arbitrarily long byte strings (its only good for short byte strings) so i'd actually expect Bytes to be something that is efficient for that use (eg base64)

Reply:
I intentionally chose this "single representation for binary blobs" to incite this discussion.  One of the aims of the redesign was to make consistent use of data formats, and taking that to its extreme involves using only one of base64 and base58 for binary data.

We could introduce two scalars again: `Digest` which is Base58 encoded, and `Base64` for binary data?  @Jordan-Mysten, thoughts on this from the perspective of having more scalars?  I would choose custom scalars over `String`s for these to make the distinction super clear.

Reply:
I don't have a strong preference. I think having a `Bytes` encoding makes some sense, e.g. you could use Uint8arrays in the client and just encode to / from base64 in transit (which makes some sense). The digest values we currently use I think should just be strings though, we never (in the client, currently at least) operate on them as binary data.

Reply:
Seems reasonable to me -- will use `String` for digests and a new `Base64` scalar for bytes.

Reply:
> I intentionally chose this "single representation for binary blobs" to incite this discussion.

Seems i took the bait then ;P